eventstore_client.go

```diff
@@ -76,15 +92,139 @@ func getStats() (*stats, error) {
 		}
 	}
 
+	var parkedMessagesStatsResult []parkedMessagesStats
+	if enableParkedMessagesStats {
+		parkedMessagesStats, err := getSubscriptionParkedMessagesStats(subscriptionsStatsResult.result)
+		if err != nil {
+			log.WithError(err).Error("Error while getting parked messages for subscriptions.")
+		} else {
+			parkedMessagesStatsResult = *parkedMessagesStats
+		}
+	}
+
 	return &stats{
 		serverStatsResult.result,
 		gossipStatsResult.result,
 		projectionStatsResult.result,
 		infoResult.result,
 		subscriptionsStatsResult.result,
+		parkedMessagesStatsResult,
 	}, nil
 }
 
+func getSubscriptionParkedMessagesStats(subscriptions []byte) (*[]parkedMessagesStats, error) {
+	var result []parkedMessagesStats
+	jp.ArrayEach(subscriptions, func(jsonValue []byte, dataType jp.ValueType, offset int, err error) {
+		eventStreamID, _ := jp.GetString(jsonValue, "eventStreamId")
+		groupName, _ := jp.GetString(jsonValue, "groupName")
+
+		lastEventNumber, err := getParkedMessagesLastEventNumber(eventStreamID, groupName)
+
+		if err != nil {
+			log.WithFields(logrus.Fields{
```

Comment:
If there was never any parked message in this subscription, the request returns HTTP 404 and a warning is logged. This situation needs to be properly handled and such subscription needs to be ignored.

Reply:
I made it return after the log, so it doesn't continue to get the tb value as its unnecessary: https://github.com/marcinbudny/eventstore_exporter/pull/14/commits/d0297a4e6c7912947714122f5a80c937856b067b